src/aruco_pose/src/aruco_pose.cpp

```diff
@@ -0,0 +1,218 @@
+  double gamma = this->get_parameter("cam_rotation.gamma").as_double() / 180 * M_PI;
+
+  // Add the camera to tf server
+  camera_quaternion_.setRPY(alpha, beta, gamma);
+
+  // Define the transform
+  geometry_msgs::msg::TransformStamped transformStamped;
+  transformStamped.header.stamp = this->now();
+  transformStamped.header.frame_id = "world";
+  transformStamped.child_frame_id = this->get_parameter("camera_tf_frame").as_string();
+  transformStamped.transform.translation.x = this->get_parameter("cam_translation.x").as_double();
+  transformStamped.transform.translation.y = this->get_parameter("cam_translation.y").as_double();
+  transformStamped.transform.translation.z = this->get_parameter("cam_translation.z").as_double();
+  transformStamped.transform.rotation.x = this->camera_quaternion_.x();
+  transformStamped.transform.rotation.y = this->camera_quaternion_.y();
+  transformStamped.transform.rotation.z = this->camera_quaternion_.z();
+  transformStamped.transform.rotation.w = this->camera_quaternion_.w();
+
+  static_broadcaster_.sendTransform(transformStamped);
+
+  camera_subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+    "/rgb/image_raw", 5,
+    std::bind(&ArucoPose::image_raw_callback, this, std::placeholders::_1));
+
+  // Assign parameters for ArUco tag detection
+  parameters_ = cv::aruco::DetectorParameters::create();
+  parameters_->cornerRefinementMethod = cv::aruco::CORNER_REFINE_APRILTAG;
+
+  auto it = dictionary_map_.find(this->get_parameter("fiducial_marker_family").as_string());
+  if (it != dictionary_map_.end()) {
+    dictionary_ = cv::aruco::getPredefinedDictionary(it->second);
+  } else {
+    RCLCPP_ERROR(
+      LOGGER, "Invalid dictionary name: %s", this->get_parameter(
+        "fiducial_marker_family").as_string().c_str());
+    throw std::runtime_error("Invalid dictionary name");
+  }
+
+  // Initial estimates
+  median_filtered_rpyxyz = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+
+  // Configure the median filter. 6 refers to the number of channels in the multi-channel filter
+  // Note: it is necessary to have/declare a parameter named 'number_of_observations'
+  // in the parameter server.
+  median_filter_->configure(
+    6, "", "number_of_observations",
+    this->get_node_logging_interface(), this->get_node_parameters_interface());
+
+  RCLCPP_INFO(LOGGER, "Pose estimator node started!");
+}
+
+void ArucoPose::image_raw_callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & rgb_msg)
+{
+  // Convert ROS image message to cv::Mat
+  cv_bridge::CvImagePtr cv_ptr_rgb =
+    cv_bridge::toCvCopy(rgb_msg, sensor_msgs::image_encodings::BGR8);
+
+  // Detect the markers from the incoming image
+  cv::aruco::detectMarkers(
+    cv_ptr_rgb->image, dictionary_, markerCorners_, markerIds_, parameters_,
+    rejectedCandidates_);
+
+  // Exclude instances where no markers are detected
+  try {
+    if (!markerIds_.empty()) {
```

Comment:
This should check for a SINGLE marker, if the functionality is limited to a single pose estimation.